tests/functional/testReplay.js

```diff
@@ -0,0 +1,241 @@
+import assert from 'assert';
+import async from 'async';
+import { Logger } from 'werelogs';
+import UtapiReplay from '../../src/lib/UtapiReplay';
+import UtapiClient from '../../src/lib/UtapiClient';
+import Datastore from '../../src/lib/Datastore';
+import redisClient from '../../src/utils/redisClient';
+import { getAllResourceTypeKeys } from '../testUtils';
+import safeJsonParse from '../../src/utils/safeJsonParse';
+const localCache = redisClient({
+    host: '127.0.0.1',
+    port: 6379,
+}, Logger);
+const datastore = new Datastore().setClient(localCache);
+const utapiClient = new UtapiClient({
+    redis: {
+        host: '127.0.0.1',
+        port: 4242, // Set the datastore client to a port it cannot connect on.
+    },
+    localCache: {
+        host: '127.0.0.1',
+        port: 6379, // Set the local cache a port for successful connection.
+    },
+});
+const log = new Logger();
+const objSize = 1024;
+// All actions supported by Utapi.
+const actions = [
+    'createBucket',
+    'deleteBucket',
+    'listBucket',
+    'getBucketAcl',
+    'putBucketAcl',
+    'putBucketCors',
+    'getBucketCors',
+    'deleteBucketCors',
+    'putBucketWebsite',
+    'getBucketWebsite',
+    'deleteBucketWebsite',
+    'uploadPart',
+    'initiateMultipartUpload',
+    'completeMultipartUpload',
+    'listMultipartUploads',
+    'listMultipartUploadParts',
+    'abortMultipartUpload',
+    'deleteObject',
+    'multiObjectDelete',
+    'getObject',
+    'getObjectAcl',
+    'putObject',
+    'copyObject',
+    'putObjectAcl',
+    'headBucket',
+    'headObject',
```

Comment:
I believe you will need to add `'getBucketLocation'` to the actions list, since the replay will check for all possible metrics and this has been added to the schema in master.